devlib/target.py

```diff
@@ -3138,6 +3138,416 @@ def _resolve_paths(self):
+        # Get CPU brand string
+        try:
+            brand_string = self.execute('sysctl -n machdep.cpu.brand_string').strip()
+        except TargetStableError:
+            brand_string = 'Unknown CPU'
+        
+        # Get CPU family, model, etc. (if available)
+        cpu_family = ''
+        cpu_model = ''
+        try:
+            cpu_family = self.execute('sysctl -n machdep.cpu.family').strip()
+        except TargetStableError:
+            pass
+        
+        try:
+            cpu_model = self.execute('sysctl -n machdep.cpu.model').strip()
+        except TargetStableError:
+            pass
+        
+        # Generate /proc/cpuinfo-like format
+        cpuinfo_text = []
+        for i in range(ncpu):
+            section = [
+                f'processor\t: {i}',
+                f'model name\t: {brand_string}',
+            ]
+            if cpu_family:
+                section.append(f'cpu family\t: {cpu_family}')
+            if cpu_model:
+                section.append(f'model\t\t: {cpu_model}')
+            
+            cpuinfo_text.append('\n'.join(section))
+        
+        return Cpuinfo('\n\n'.join(cpuinfo_text))
+
+    @property
+    def hostname(self):
+        """Get hostname using appropriate command for the platform"""
+        if self._is_macos():
+            # macOS has native hostname command
+            return self.execute('hostname').strip()
+        else:
+            # Use parent implementation for Linux systems
+            return super(LocalTarget, self).hostname
+
+    @property
+    @memoized  
+    def kernel_version(self):
+        """Get kernel version using appropriate command for the platform"""
+        if self._is_macos():
+            # macOS has native uname command  
+            return KernelVersion(self.execute('uname -r -v').strip())
+        else:
+            # Use parent implementation for Linux systems
+            return super(LocalTarget, self).kernel_version
+
+    @property
+    def hostid(self):
+        """Get host ID using appropriate method for the platform"""
+        if self._is_macos():
+            try:
+                return int(self.execute('hostid').strip(), 16)
+            except (TargetStableError, ValueError):
+                # hostid might not be available on macOS, return a default
+                return 0
+        else:
+            # Use parent implementation for Linux systems
+            return super(LocalTarget, self).hostid
+
+    @asyn.asyncf
+    async def write_value(self, path, value, verify=True, as_root=True):
+        """Write value to file using platform-appropriate command"""
+        if self._is_macos():
+            # macOS has native printf
+            self.async_manager.track_access(
+                asyn.PathAccess(namespace='target', path=path, mode='w')
+            )
+            value = str(value)
+
+            if verify:
+                # Use the same verification logic as the parent class but with native commands
```

Comment:
> but with native commands

What is the advantage for that ? The main reason we use busybox everywhere in devlib is so that we know what behaviour to expect. While a mac target has to deal with only one OS rather than multiple for linux-based code, there may still be changes from one version to another as we have painfully learnt with android over the years.